lib/dry/validation/rule/attr.rb

```diff
@@ -0,0 +1,21 @@
+module Dry
+  module Validation
+    class Rule::Attr < Rule
+      def self.new(name, predicate)
+        super(name, predicate.curry(name))
+      end
+
+      def type
+        :attr
+      end
+
+      def evaluate_input(input)
+        input.public_send(name)
+      end
+
+      def call(input)
+        Validation::Rule::Result::EvaluatedValue.new(input, predicate.(input), self)
```

Comment:
We discussed it here: https://github.com/dryrb/dry-validation/pull/32#discussion_r48425294
It will work in a way that it does `input.public_send(name)` call only when `attr?` predicate returns success, so we will prevent `NoMethodError` to be triggered when `input` doesn't respond to specific attribute expected in schema.
Here is the place where I do this check and call attribute:  https://github.com/dryrb/dry-validation/pull/29/files#diff-f8751eafe5dec4fd9321589ed3e2b792R68